protocols/gossipsub/src/lib.rs

```diff
@@ -125,7 +125,7 @@ pub use self::subscription_filter::{
 };
 pub use self::topic::{Hasher, Topic, TopicHash};
 pub use self::transform::{DataTransform, IdentityTransform};
-pub use self::types::{Message, MessageAcceptance, MessageId, RawMessage, Rpc};
```

Comment:
`Rpc` doesn't need to be public right?

Reply:
I dont think so

Reply:
I thought it is mentioned in some transform API or something? You should be able to test by making it `pub(crate)`.

Note that that is a breaking change so we can't do it right away.

Reply:
oh yeah good point. The compression algo prob needs it.

Reply:
I just tried and it can be made private! :tada:

Reply:
> The compression algo prob needs it.

`RawMessage` could be private too if we'd deprecate `DataTransform` in favor of an `fn(Message) -> Message` right? Would that be an option or do you necessarily need the data that is in `RawMessage` but not `Message`?